### ISSUE TITLE
offline mode practically unusable in MAC devices

### ISSUE THREAD

**Issue by panosnethood** [mazi-project/portal#90]
Hi, I used recently the offline mode in two real situations and I share here my experiences hoping that we will solve soon the issues with MAC devices in this mode.

So, I had stopped since long time to use the offline mode (I thought that the "dual" mode communicating an "easy" handmade URL works better) but recently I did try it successfully in a big event where I chose only the guestbook as the splash page (both to focus people on this application but also because of the issues mentioned in #40 and #67). The captive portal worked very nicely with all devices and especially with iPhones,  people were very happy and impressed with the "automatic" launch of the guestbook (but this wouldn't be the case for nextcloud, interview app, etc). 

So, yesterday I gave a course for "advanced users" on the mazi toolkit and thought to start from the offline mode, since the idea was to train people to set-up the mazi toolkit from the beginning and this is the default mode anyway. Half of the people had macs and the pop-up window that appeared when the connected to the MAZI Zone could not be maximized and it was practically unusable, and everyone was shouting "it does not work". It was very embarrassing for me, I had to give the URL, people didn't type it correctly, it was a mess.

In short, I don't see a scenario with "real" people in which the offline mode would work properly (except if applications are restricted to guestbook + etherpad) for MAC devices, and these are the most popular ones.

So, I think that we should either correct this or remove completely the offline mode, because to explain properly in the guidelines (not to mention in a public place for users that have never seen before the mazi toolkit) gets really messy and will not look very nice. "If this, and if that, etc."

**Comment by panosnethood** [comment]
What I propose to address this issue is the following: If it is not easy to solve this let's say until the end of the month, we should change the default behaviour of the MAZI Zone and make the "dual" option as the default, thinking a good "local URL strategy" and making "offline mode" more "experimental" explaining properly to those that want to use it the restrictions that exist.

**Comment by haniavis** [comment]
We are working on it and there will be a new version soon featuring also other bug fixes.

The approach we are working is to have an initial information page in this light-browser appearing in MAC devices, which will inform users about the mazizone and they will need to click a button to proceed. This way, the light-browser will disappear and the user will be able to use the proper browser.

**Comment by panosnethood** [comment]
This would be really great! Thanks for the quick reply and sorry for insisting on this but it concerns the very first contact with the MAZI Zone and it is really important to have a clear strategy, also reflected in the guidelines (right now they are misleading since they don't explain well this "glitch"). 

And some more thoughts on this: it would be perhaps meaningful to create an "first contact" page for all environments (independently from device) that would be displayed only the first time someone connects to a MAZI Zone and then never appear again (for the same MAC address or something).

It could be something like this: "You are connected to a local network managed by XX. [Perhaps some statistics] Click OK to continue"

**Comment by mgaved** [comment]
Hi Panos - 
_"it would be perhaps meaningful to create a "first contact" page for all environments (independently from device) that would be displayed only the first time someone connects to a MAZI Zone and then never appear again "_

Are you suggesting a 'splash screen' like those that you find in e.g. hotels when you sign on to the hotel wifi for the first time? (and after this, you do not see the hotel's home page). I think James (SPC) has also raised this suggestion.

**Comment by haniavis** [comment]
Yes this is similar to what we were discussing in our last face-to-face meeting.

**Comment by haniavis** [comment]
#67

**Comment by panosnethood** [comment]
Thanks Harris! I think that the next solution you describe above is still very important and urgent because otherwise we will need to change the guidelines to have different "options" based on one's device and this is not good.

**Comment by panosnethood** [comment]
Btw, if this is not resolved soon, we should also offer somehow the option of the "aggressive" captive portal because for simpler applications (e.g., guestbook) having the automatic window pop-up in MAC is very useful. I am working actually with some people that are very interested in this case ...

**Comment by panosnethood** [comment]
Trying the new captive portal functionality in a MAC computer, and when one clicks on the photo to enter the portal this happens inside this restricted pop-up window. I hoped that with nodogsplash the link would open a normal browser ... any hints?
